echopype/convert/api.py

```diff
@@ -428,20 +428,16 @@ def open_raw(
     # Check file extension and existence
     file_chk, xml_chk = _check_file(raw_file, sonar_model, xml_path, storage_options)
 
-    # TODO: the if-else below only works for the AZFP vs EK contrast,
-    #  but is brittle since it is abusing params by using it implicitly
-    if SONAR_MODELS[sonar_model]["xml"]:
-        params = xml_chk
-    else:
-        params = "ALL"  # reserved to control if only wants to parse a certain type of datagram
+    params = xml_chk
 
     # obtain dict associated with directly writing to zarr
     dgram_zarr_vars = SONAR_MODELS[sonar_model]["dgram_zarr_vars"]
 
     # Parse raw file and organize data into groups
     parser = SONAR_MODELS[sonar_model]["parser"](
         file_chk,
-        params=params,
+        # Currently used only for AZFP XML file
+        file_meta=params,
```

Comment:
How about just changing this to `file_meta=xml_chk` and the line (`params = xml_chk`) above?

Reply:
Updated!